### PR TITLE
refactor(cheatsheet):added tests and changed some logic

### PR DESCRIPTION
**Pull request by Her0elt**
# refactor cheatsheet added tests and changed some logic
## Refactored cheatsheet logic and viewset and added integration test fro the cheatsheet viewset

Comments/issues:
related to refactor cheatsheet in tihlde/Kvark#285
this pr liv result inn a problem with everyone in the database that is registered as diginc because its change to digsec in this pr

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] PR name is descriptive
- [ ] The PR includes the link to the relevant Trello task
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

- [ ] `python manage.py runserver` was run locally witout errors

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Review by Eirsteir — CHANGES_REQUESTED** [review]
Digging the testing train you're on! 

While you are at it, could you translate to return messages to norwegian?

File: app/content/urls.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 )
 router.register("notification", NotificationViewSet, basename="notification")
 router.register(
-    r"cheatsheet/study/(?P<study>[^\.]+)/grade/(?P<grade>[^\.]+)/file",
+    r"cheatsheet/(?P<study>[^\.]+)/(?P<grade>[^\.]+)/files",
```
[the remark this answers is not in the document]
Comment (Zenjjim) [reply app/content/urls.py:36]: Tror det er bedre om det er mulig å fjerne files

**Review by olros — APPROVED** [review]
🗒️  📓  📔  🎶

File: app/content/urls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eirsteir) [reply app/content/urls.py:34]: Enig i den forsåvidt, fikser du det før du merger @Her0elt? 😄 

@olros da må nok frontend oppdatere url-en.

**Review by Eirsteir — APPROVED** [review]
Please update the endpoint url before you merge 😊

**Pull request merged**